lua/entities/prop_vehicle_crawler/cl_init.lua

```diff
@@ -0,0 +1,209 @@
+local wheel_mat = Material("models/crawler/energy_wheel")
+
+local max_bar_height = 256 - 100
+local max_velocity = 3000
+
+local gizo_cam_pos = Vector(-460, -460, 650)
+local gizmo_cam_ang = Angle(45, 45, 0)
+local boost_text_col = Color(255, 0, 0)
+local WHEEL_OFFSET = Vector(0, 0, 10)
+
+hook.Add("CalcVehicleView", tag, function(veh, ply, view)
+	local crawler = veh:GetParent()
+
+	if not IsValid(crawler) then return end
+	if crawler:GetClass() ~= tag then return end
+	if not veh:GetThirdPersonMode() then return end
+
+	local tr = util.TraceHull( {
+		start = view.origin,
+		endpos = view.origin - view.angles:Forward() * (50 + veh:GetCameraDistance() * 50),
+		filter = {crawler, veh, ply},
+		mins = Vector(-4, -4, -4),
+		maxs = Vector(4, 4, 4),
+	} )
+
+	view.drawviewer = true
+	view.origin = tr.HitPos
+	return view
+end)
+
+function ENT:SetupWheel()
+	self.Wheel = ClientsideModel("models/crawler/energy_wheel.mdl", RENDERGROUP_BOTH)
+	self.Wheel:SetPos(self:LocalToWorld(WHEEL_OFFSET))
+	self.Wheel:SetAngles(self:LocalToWorldAngles(Angle(0, 0, 0)))
+	self.Wheel:SetColor(self.EnergyColor)
+	self.Wheel.RenderGroup = RENDERGROUP_BOTH
+	self.Wheel:SetParent(self)
+	self.Wheel:Spawn()
+end
+
+function ENT:Initialize()
+	self.vel_increment = 0
+	self.steering_wheel_angle = 0
+
+	self:SetupWheel()
+
+	self.DashboardTexture = self.DashboardTexture or GetRenderTargetEx(
+		"Crawler_Dashboard_" .. self:EntIndex(),
+		512,
+		256,
+		RT_SIZE_LITERAL,
+		MATERIAL_RT_DEPTH_NONE,
+		16,
+		CREATERENDERTARGETFLAGS_HDR,
+		IMAGE_FORMAT_DEFAULT
+	)
+	self.DashboardMaterial = self.DashboardMaterial or CreateMaterial(
+		"Crawler_Dashboard_Material_" .. self:EntIndex(),
+		"VertexLitGeneric",
+		{
+			["$basetexture"] = self.DashboardTexture:GetName(),
+			["$model"] = 1,
+			["$nodecal"] = 1,
+			["$selfillum"] = 1,
+			["$selfillummask"] = "dev/reflectivity_30b"
+		}
+	)
+
+	local clr = self:GetEnergyColor()
+	self.EnergyColor = Color(clr.x, clr.y, clr.z)
+	self.GizmoModel = self.GizmoModel or ClientsideModel("models/crawler/gizmo.mdl")
+	self.GizmoModel:SetNoDraw(true)
+
+end
+
+local Ride_Height_Visual = Ride_Height + 11
+function ENT:Think()
+	if IsValid(self.Wheel) then
+		--Wheel spin
+		self.vel_local = self:WorldToLocal(self:GetPos() + self:GetVelocity() * 180 * FrameTime())
+		self.vel_increment = self.vel_increment  + (self.vel_local[1] / 295.30970943744) --circumference of the wheel
+		self.Wheel:SetAngles(self:LocalToWorldAngles(Angle(self.vel_increment, 0, 0)))
+
+		--Wheel suspension
+		local Terrain_Distance = util.TraceHull({
+			start = self:LocalToWorld(WHEEL_OFFSET),
+			endpos = self:LocalToWorld(WHEEL_OFFSET - Vector(0, 0, Ride_Height_Visual)),
+			filter = self,
+			mins = Vector(-2, -2, -2),
+			maxs = Vector(2, 2, 2),
+			mask = MASK_SOLID,
+			collisiongroup = COLLISION_GROUP_WEAPON
+		})
+		self.Wheel:SetPos(self:LocalToWorld(WHEEL_OFFSET + Vector(0, 0, Ride_Height_Visual- Terrain_Distance.Fraction * Ride_Height_Visual)))
```

Comment:
Some spaces missing between operators, put the files through a formatter / linter of some kind